tests/src/test/java/org/mozilla/javascript/tests/Utils.java

```diff
@@ -91,37 +77,44 @@ public static boolean isJavaVersionAtLeast(int desiredVersion) {
     }
 
     public static void assertWithAllOptimizationLevels(final Object expected, final String script) {
-        runWithAllOptimizationLevels(
-                cx -> {
-                    final Scriptable scope = cx.initStandardObjects();
-                    final Object res = cx.evaluateString(scope, script, "test.js", 0, null);
-
-                    assertEquals(expected, res);
-                    return null;
-                });
+        assertWithAllOptimizationLevels(-1, expected, script);
     }
 
     public static void assertWithAllOptimizationLevels_1_8(
             final Object expected, final String script) {
-        runWithAllOptimizationLevels(
-                cx -> {
-                    cx.setLanguageVersion(Context.VERSION_1_8);
-                    final Scriptable scope = cx.initStandardObjects();
-                    final Object res = cx.evaluateString(scope, script, "test.js", 0, null);
-
-                    assertEquals(expected, res);
-                    return null;
-                });
+        assertWithAllOptimizationLevels(Context.VERSION_1_8, expected, script);
     }
 
     public static void assertWithAllOptimizationLevelsES6(
             final Object expected, final String script) {
+        assertWithAllOptimizationLevels(Context.VERSION_ES6, expected, script);
+    }
+
+    public static void assertWithAllOptimizationLevels(
+            final int languageVersion, final Object expected, final String script) {
         runWithAllOptimizationLevels(
                 cx -> {
-                    cx.setLanguageVersion(Context.VERSION_ES6);
+                    if (languageVersion > -1) {
```

Comment:
I see, `-1` does mean "do not change version" here.

Reply:
yes maybe i have to add some comment....